crates/matrix-sdk/src/sliding_sync/list/mod.rs

```diff
@@ -322,28 +264,10 @@ impl SlidingSyncListInner {
     /// request will be sent and a response will be received. The
     /// [`Self::maximum_number_of_rooms`] won't change.
     pub fn set_sync_mode(&self, sync_mode: SlidingSyncMode) {
-        // Acquire both locks before modifying the values their hold.
-        let mut sync_mode_lock = self.sync_mode.write().unwrap();
-        let mut request_generator_lock = self.request_generator.write().unwrap();
-
-        // Now we can modify both values.
-        *sync_mode_lock = sync_mode.clone();
-        *request_generator_lock = SlidingSyncListRequestGenerator::new(sync_mode);
-    }
-
-    /// Reset and add new ranges.
-    fn set_ranges(&self, ranges: &[RangeInclusive<Bound>]) {
-        *self.ranges.write().unwrap() = ranges.to_vec();
-    }
-
-    /// Add a new range.
-    fn add_range(&self, range: RangeInclusive<Bound>) {
-        self.ranges.write().unwrap().push(RangeInclusive::new(*range.start(), *range.end()));
-    }
-
-    /// Call this method when it's necessary to reset `Self`.
-    fn reset(&self) {
-        self.request_generator.write().unwrap().reset();
+        {
```

Comment:
So now, changing the `sync_mode` resets the state of the list too?

Please update the comment at line 263, that say the opposite :-).